### PR TITLE
Lighning-input compression for mobile

### PR DESCRIPTION
**Pull request by albarivas**
### What does this PR do?
Fix missing LWC if/else directives
Adds example of using the new mediaUtils library to compress and resize images (especially suitable for mobile)

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: force-app/main/default/classes/FileUtilities.cls [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        String base64data,
+        String filename,
+        String recordId
+    ) {
+        try {
+            ContentVersion contentVersion = new ContentVersion();
+            contentVersion.VersionData = EncodingUtil.base64Decode(base64data);
+            contentVersion.Title = filename;
+            contentVersion.PathOnClient = filename;
+            insert contentVersion;
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilities.cls:13]:
```suggestion
            insert as user contentVersion;
```

File: force-app/main/default/classes/FileUtilities.cls [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                FROM ContentVersion
+                WHERE Id = :contentVersion.Id
+                WITH USER_MODE
+            ];
+
+            ContentDocumentLink contentDocumentLink = new ContentDocumentLink();
+            contentDocumentLink.ContentDocumentId = contentVersion.ContentDocumentId;
+            contentDocumentLink.LinkedEntityId = recordId;
+            contentDocumentLink.ShareType = 'V';
+            insert contentDocumentLink;
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilities.cls:26]:
```suggestion
            insert as user contentDocumentLink;
```

File: force-app/main/default/classes/FileUtilities.cls-meta.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>56.0</apiVersion>
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilities.cls-meta.xml:3]:
```suggestion
    <apiVersion>57.0</apiVersion>
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        String recordId = 'INVALID_ID';
+
+        try {
+            // WHEN
+            String contentDocumentLinkId = FileUtilities.createFile(
+                base64Data,
+                fileName,
+                recordId
+            );
+            Assert.fail();
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:38]:
```suggestion
            Assert.fail('Expected an AuraHandledException');
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        } catch (Exception e) {
+            // THEN
+            Assert.isTrue(e instanceof AuraHandledException);
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:41]:
```suggestion
            Assert.isInstanceOfType(e, AuraHandledException.class);
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        String recordId = property.Id;
+
+        try {
+            // WHEN
+            String contentDocumentLinkId = FileUtilities.createFile(
+                base64Data,
+                fileName,
+                recordId
+            );
+            Assert.fail();
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:62]:
```suggestion
            Assert.fail('Expected an AuraHandledException');
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        } catch (Exception e) {
+            // THEN
+            Assert.isTrue(e instanceof AuraHandledException);
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:65]:
```suggestion
            Assert.isInstanceOfType(e, AuraHandledException.class);
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        String recordId = property.Id;
+
+        try {
+            // WHEN
+            String contentDocumentLinkId = FileUtilities.createFile(
+                base64Data,
+                fileName,
+                recordId
+            );
+            Assert.fail();
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:86]:
```suggestion
            Assert.fail('Expected an AuraHandledException');
```

File: force-app/main/default/classes/FileUtilitiesTest.cls [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        } catch (Exception e) {
+            // THEN
+            Assert.isTrue(e instanceof AuraHandledException);
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls:89]:
```suggestion
            Assert.isInstanceOfType(e, AuraHandledException.class);
```

File: force-app/main/default/classes/FileUtilitiesTest.cls-meta.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>56.0</apiVersion>
```
Comment (pozil) [inline force-app/main/default/classes/FileUtilitiesTest.cls-meta.xml:3]:
```suggestion
    <apiVersion>57.0</apiVersion>
```

**Review by pozil — CHANGES_REQUESTED** [review]
Made some suggestions

File: force-app/main/default/classes/FileUtilities.cls (hunk 1 above)
Comment (albarivas) [reply force-app/main/default/classes/FileUtilities.cls:13]: I had this, but removed it because I couldn't commit - any updates on the eslint plugin fix?

Comment (pozil) [reply force-app/main/default/classes/FileUtilities.cls:13]: Let's proceed without this for now (see my Slack message).

**Review by pozil — APPROVED** [review]
Good to go. Thanks @albarivas

**Review by pozil — APPROVED** [review]
Looking good, thanks @albarivas

**Pull request merged**